backend/director/agents/web_search_agent.py

```
@@ -0,0 +1,219 @@
+            f"search_videos: {search_videos}"
+        )
+
+        search_engine_tool = None
+        if engine == "serp":
+            self.api_key = os.getenv("SERP_API_KEY")
+            if not self.api_key:
+                return AgentResponse(
+                    status=AgentStatus.ERROR,
+                    message="SERP_API_KEY environment variable is not set.",
+                )
+            serp_config = search_videos.get("serp_config", {})
```

Ruff (0.8.2)

Local variable `serp_config` is assigned to but never used

Remove assignment to unused variable `serp_config`

(F841)